lib/rubocop/cop/rails/timeout_from_transaction.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'rubocop'
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks for transactions wrapped in timeouts,
+      # motivated by change in Rails behavior that will
```

Comment:
It would be even more user-friendly with a few more summaries discussed in the PR https://github.com/rails/rails/pull/29333.